### PR TITLE
docs(plan): port i18n no-literal-string doctrine and catalogue reference (#1441)

### PR DESCRIPTION
**Pull request by lhpaul**
## Summary

Implementation plan for #1441 (Refactor path — tracker brief only, no
product spec). Ships as documentation only: a new
`docs/best-practices/stack/i18n.md` worked reference (portable doctrine +
React Native/i18next worked example + the dynamic-key scanner pitfall named
in the issue), a conditional `REVIEW.md` checklist block, and two small
cross-references. No executable code is added — see the plan's
**Template-Fit Assessment** section for the reasoning.

## Template-Fit Assessment

This repository ships no application source tree and no JS/TS test runner.
Porting the source PR's `eslint-plugin-i18next` config and Jest suites
verbatim as live files here would create orphaned, never-executed code —
itself a declared-but-unverified control. The plan narrows scope to
documentation (Protocol 02 Step 0 option 2), following the explicit
narrowing framing already supplied in this item's own dispatch
instructions. Full three-bucket breakdown (portable doctrine / stack-specific
worked example / not portable at all) is in the plan.

## Document Quality Gate

- Spec/brief coverage: Checked - all three brief asks (lint rule + test
  suite discipline, catalogue/resolver skeleton, protocol note) map to plan
  sections and Implementation Order steps, scoped per the Template-Fit
  Assessment.
- Implementation-order consistency: Checked - file list (i18n.md,
  STACK-SPECIFIC.md, 3-testing.md, REVIEW.md, CHANGELOG.md) and Implementation
  Order agree; the plan PR itself contains only the plan and smoke-test
  runbook, no implementation artifacts.
- Verification support: Checked - all broad claims (no app source tree, no
  JS/TS test runner, stack/ doc-only precedent, source PR proof transcript,
  REVIEW.md block format) cite Verification Log commands/queries.
- Complex workflow decision-gate matrix: Not applicable - this plan does not
  add or modify workflow decision-gate behavior.
- Parser/API/concurrency checklist: Not applicable - no parser, API-surface,
  snapshot, or concurrent-event signals; the scanner is reproduced only as
  an illustrative doc code block, not an executed file (see plan's
  "Parser-risk classification note").

## WHY_SAFE_TO_MERGE

- **scope**: Adds only the implementation plan
  (`docs/specs/developments/20260811002516_1441-i18n-lint-rule-port/2_1441-i18n-lint-rule-port_implementation-plan.md`)
  and its plan-stage smoke-test runbook
  (`docs/testing/workflow/1441-i18n-lint-rule-port.smoke-test.md`). No
  implementation files, no `CHANGELOG.md` entry (plan branches are exempt),
  no other repository surface touched.
- **tests**: N/A for a plan-only PR — no code is added. Both repository
  markdown lint passes (`markdownlint-cli2` and
  `scripts/lint/markdown-heuristic-lint.py`) and the shell-snippet lint
  (`scripts/lint/workflow-shell-snippet-lint.py --base-ref origin/develop`)
  were run locally against the two new files and pass with exit code 0.
- **reviewer_outcome**: pending — Step 7a / Step 7 to run after this PR is
  opened.
- **ci_outcome**: pending — final head SHA to be recorded once CI runs on
  the pushed commit.
- **rollback_or_cleanup_risk**: Minimal. Reverting deletes two new
  documentation files under `docs/specs/developments/` and
  `docs/testing/workflow/`; nothing else references them yet, and no
  downstream tooling consumes this plan file.

## Summary by CodeRabbit

* **Documentation**
  * Added an implementation plan for porting internationalization guidance, including React Native/i18next examples, scanner edge cases, verification steps, risks, and testing considerations.
  * Added a smoke-test runbook covering prerequisites, validation procedures, dynamic-key handling, review checks, troubleshooting, and known limitations.

### CODE REVIEW

**Comment by lhpaul** [comment]
WARNING: internal_reviewer 'cursor' unreachable from current runner (Claude Code subagent) — skipping. Only 'claude' will run in this Step 7a cycle. Reviewer coverage is reduced from 2 to 1.

**Comment by lhpaul** [comment]
## Step 7a — Internal Plan Review (Protocol 02 / REVIEW.md Plan Review Checklist)

**Verdict: APPROVED**

### Scope verification
- PR diff contains exactly two files: the implementation plan and the plan-stage smoke-test runbook. No implementation files, no `CHANGELOG.md` change (plan branches are exempt). Confirmed via `gh pr view --json files`.

### Template-Fit Check (Protocol 02 Step 0) — independently verified, not just trusted
- Confirmed no `apps/`, `src/`, or product application source tree exists anywhere in the repo (`find . -maxdepth 2 -type d`).
- Confirmed `docs/best-practices/stack/supabase.md` is the established doc-only precedent: pure prose + illustrative, non-executed TypeScript code blocks — matches how this plan proposes to ship `i18n.md`.
- Confirmed `docs/best-practices/STACK-SPECIFIC.md` already carries a concrete `| Supabase | [stack/supabase.md](stack/supabase.md) |` row alongside placeholder rows — the new i18n row follows the same pattern.
- Confirmed source reference `lhpaul/personal-finances` PR #41 is MERGED (`gh pr view 41 --repo lhpaul/personal-finances`).
- Confirmed the plan's cited protocol option ("Protocol 02 Step 0 option 2" — narrow scope) is a real, numbered option in `02-generate-implementation-plan-protocol.md` Step 0, not an invented citation.
- **Found and fixed a factual inaccuracy** in the Summary and Template-Fit Assessment sections: the plan originally claimed "no JS/TS test runner" and "no JS/TS files under version control anywhere except node_modules/build artifacts." This is false and contradicted the plan's own Verification Log row — `e2e/` ships a Playwright E2E runner (`e2e/package.json` → `"test": "playwright test"`) with a tracked placeholder spec (`e2e/tests/baseline.spec.ts`), and `hooks/` ships several tracked `.ts` files (Haystack git-hook internals). Corrected both sections to the accurate, narrower, and still load-bearing claim: no application source tree and no unit-test runner (Jest/Vitest) capable of hosting the source PR's `.test.ts` suites. The doc-only conclusion itself is unaffected — Playwright E2E tooling and Haystack hook internals are not a home for ported i18n unit tests either way.
- With that correction, the narrowing to documentation-only is well-justified and independently verifiable against the actual repo structure, not just the plan's own say-so.

### Other checklist items
- `Cross-Cutting Operational Assumption Check` present with a reasoned `Not applicable` result (no shared environment/cloud/base-branch/config dependency; batch-context cross-check against concurrent items is explicit).
- `Document Quality Gate` log in the PR description matches the plan's actual content (spec/brief coverage, implementation-order consistency, verification support, decision-gate matrix N/A, parser/concurrency checklist N/A — all accurate on inspection).
- Verification Log entries reproduced and confirmed: `REVIEW.md`'s existing "Additional checks for" blocks count (10, confirmed via grep) and the cited insertion point (line ~318, confirmed), the `STACK-SPECIFIC.md` Supabase-row precedent, the `supabase.md` doc-only pattern, and the source PR's merged state.
- CHANGELOG literal in Implementation Order step 6 uses the project's `**Bold Title** (#N):` format, not conventional-commit format — confirmed against existing `CHANGELOG.md` entries.
- Ran the plan's own validation commands locally against the two new files: `markdownlint-cli2` → 0 errors; `markdown-heuristic-lint.py` → exit 0; `workflow-shell-snippet-lint.py --base-ref origin/develop` → exit 0.

### Fix applied
- Commit `518626f` — "fix(plan): correct JS/TS toolchain claim in Template-Fit Assessment" (pushed to `implementation-plan/1441-i18n-lint-rule-port`).

### Issues reported (not fixed — none required a human decision)
- None. No blocking or important findings remain after the fix above.

**Comment by lhpaul** [comment]
@coderabbitai review

**Comment by lhpaul** [comment]
### Automated Reviewer Loop Summary

**Result:** clean — no blocking findings
**Platforms:** pr-agent (clean), coderabbit (clean)
**Findings:** 0 blocking, 0 suggestions

**Ready reviewer phase:** No net-new blocker was found after the draft GitHub gate.
**Ready-phase platforms:** coderabbit

**Advisory findings (non-blocking):**
- Possible Issue ([view comment](https://github.com/lhpaul/ai-dev-framework-template/pull/1466#issuecomment-5247619120))

*Posted automatically by `pr-review-loop.sh`.*

<details>
<summary>Reviewer-loop history (4 iterations)</summary>


```json
{
  "schema": "reviewer_loop_history.v1",
  "pr_number": 1466,
  "updated_at": "2026-08-11T01:09:28Z",
  "history_status": "available",
  "history_unavailable_reason": "",
  "entries": [
    {
      "iteration": 1,
      "recorded_at": "2026-08-11T00:36:16Z",
      "head_sha": "518626f68c4cf4bd7df8ba4b65189c0f7cb3961e",
      "result": "clean",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "coderabbit (clean)"
      ],
      "blocking_count": 0,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "coderabbit"
        ],
        "started": true,
        "net_new_blocker": false,
        "blocking_platform": ""
      }
    },
    {
      "iteration": 2,
      "recorded_at": "2026-08-11T00:49:29Z",
      "head_sha": "a19cbe57bb70985965ef3951fbe025f975dba274",
      "result": "clean",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "coderabbit (clean)"
      ],
      "blocking_count": 0,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "coderabbit"
        ],
        "started": true,
        "net_new_blocker": false,
        "blocking_platform": ""
      }
    },
    {
      "iteration": 3,
      "recorded_at": "2026-08-11T00:59:49Z",
      "head_sha": "a57e4019a5b107d2b70cc06a8bcc29e86b00c29f",
      "result": "clean",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "coderabbit (clean)"
      ],
      "blocking_count": 0,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "coderabbit"
        ],
        "started": true,
        "net_new_blocker": false,
        "blocking_platform": ""
      }
    },
    {
      "iteration": 4,
      "recorded_at": "2026-08-11T01:09:28Z",
      "head_sha": "25aa123692aa5d296eb3863f28bf6e4279fdf8fe",
      "result": "clean",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "coderabbit (clean)"
      ],
      "blocking_count": 0,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "coderabbit"
        ],
        "started": true,
        "net_new_blocker": false,
        "blocking_platform": ""
      }
    }
  ]
}
```
</details>

**Comment by lhpaul** [comment]
Addressed all 6 CodeRabbit findings from the ready-phase review in commit a19cbe5:

1. **L100 verification-log inconsistency** — replaced the limited top-level `find` with a repo-wide `git ls-files -- '*.ts' '*.tsx'` scan that proves the tracked-file claim at any depth, and corrected the 'no committed spec files' claim to accurately describe the tracked `e2e/tests/baseline.spec.ts` placeholder.
2. **L367 fail-closed lint pipeline** — replaced the bare `find | xargs` with a fail-closed `bash` block that validates both scan directories exist, validates the discovered file list is non-empty, and propagates `find`/`python3` failures (uses a Bash-3.2-compatible while-read loop, not the Bash-4-only `mapfile`).
3. **L104 unsupported git-history claim** — added the exact `git log --follow --oneline -- apps/mobile/src/test-utils/catalogue-key-scan.ts` command and its actual output (one commit, `54d4390`) to the verification log.
4. **L267 lint-coverage conflation** — split the Testing Strategy scenario into standard `markdownlint-cli2` coverage (all six files) vs. the heuristic lint's narrower `docs/specs/developments/` + `docs/testing/workflow/` scope.
5. **L366 missing plan file in markdownlint-cli2** — added the implementation plan file itself to the lint command; it was omitted.
6. **smoke-test.md L176-177 invalid troubleshooting cause** — removed 'new file added outside the linted glob patterns' as a stated cause of a lint failure on a file the linter never processes.

All fixes verified locally: `markdownlint-cli2`, the heuristic lint script (now fail-closed), and `workflow-shell-snippet-lint.py --base-ref origin/develop` all pass with exit code 0 on both changed files.

All 6 review threads resolved.

**Comment by lhpaul** [comment]
@coderabbitai review

**Comment by lhpaul** [comment]
## /run-epic PR Disposition

- Scope source: run-items-batch (sub-batch 10 of 11, tool-fix serialize-first)
- Item: #1441 - Port the no-literal-string i18n lint rule and catalogue skeleton from personal-finances
- PR: #1466
- Reviewed head SHA: `25aa123692aa5d296eb3863f28bf6e4279fdf8fe`
- Reviewer result: clean
- Blocking findings: 0
- Advisory findings: 0
- Risk: low
- Merge authority: delegated (stages.plan.may_merge_pr=true, max_merge_risk=medium; actual risk=low)
- Final decision: ready-for-human-review (parent runner scoped to opening/readying PRs only; Gate 5 delegated merge decision and merge execution belong to the parent /run-items batch runner, not this item runner)

### Risk Reasons

- Plan-only PR (implementation-plan/1441-i18n-lint-rule-port) adding two new documentation files under docs/specs/developments/ and docs/testing/workflow/; no product code, no CI, no executable check is added.
- Documentation-stage alignment checker confirms the PR contains only expected plan-stage artifacts.
- Central Template-Fit Assessment decision (documentation-only port, not executable code) was independently re-verified by the Step 7a implementation-plan-reviewer against the actual repo structure, not just trusted from the plan's own claims.

### Why Safe to Merge

- Scope: Adds only the implementation plan (docs/specs/developments/20260811002516_1441-i18n-lint-rule-port/2_1441-i18n-lint-rule-port_implementation-plan.md) and its plan-stage smoke-test runbook (docs/testing/workflow/1441-i18n-lint-rule-port.smoke-test.md). No implementation files, no CHANGELOG.md entry (plan branches are exempt), no other repository surface touched. Documentation-stage alignment checker confirms: DOCUMENTATION_STAGE_ALIGNMENT=aligned.
- Tests: N/A for a plan-only PR - no code is added. markdownlint-cli2, scripts/lint/markdown-heuristic-lint.py (now fail-closed), and scripts/lint/workflow-shell-snippet-lint.py --base-ref origin/develop all pass locally with exit code 0 on both changed files.
- Reviewer outcome: Step 7a: implementation-plan-reviewer (claude) APPROVED after one fix. cursor was unreachable from this runner context and skipped per warn policy. Step 7: four review rounds total, each caught by a direct GraphQL isResolved check (not just the loop script's self-report), since CodeRabbit's late/async passes repeatedly slipped past the loop's 30s post-clean recheck window. Round 1: 6 findings (verification-log accuracy, fail-closed lint pipeline, missing plan file in markdownlint-cli2, lint-scope wording) - fixed and resolved. Round 2: 2 findings (unsupported package-manifest claim, process-substitution masking risk) - fixed and resolved. Round 3: 3 findings (git-ls-tree determinism + JS/TS scope narrowing, stale pipefail/process-substitution prose, and a CHANGELOG.md-in-heuristic-lint-command finding that was investigated and declined with cited evidence from scripts/lint/README.md and the script's own docstring, then explicitly resolved with that rationale) - 2 fixed, 1 explained and resolved. Round 4 (final): pr-agent clean (one acknowledged non-blocking 'Possible Issue' advisory about comment wording), coderabbit clean, 0 unresolved threads confirmed via 6 direct GraphQL polls over 3+ minutes.
- CI outcome: Green - final head SHA 25aa123692aa5d296eb3863f28bf6e4279fdf8fe. All 6 checks: Markdown lint SUCCESS, PR policy SUCCESS, PR-Agent review SUCCESS, CodeRabbit SUCCESS, Reviewer-loop completion guard SUCCESS, E2E regression (placeholder) SKIPPED (template placeholder, not applicable to a docs-only PR).
- Rollback or cleanup risk: Minimal. Reverting deletes two new documentation files under docs/specs/developments/ and docs/testing/workflow/; nothing else references them yet, and no downstream tooling consumes this plan file. No schema, config, or code changes to roll back.

### Invocation Policy

Not recorded.

### Checkpoint Policy

Not recorded.

### Advisory Decisions

None.

### Verification Evidence

- Labels: 
- CI result: unknown
- Reviewer summary present: false
- Unresolved thread count: unknown
- PR merge state: unknown
- Issue state: unknown
- Project status: unknown

### Protocol Deviations

None.

**Comment by lhpaul** [comment]
Addressed the 2 additional CodeRabbit findings from the follow-up review round in commit a57e401:

1. **L100 unsupported package-manifest claim** — added the exact commands (`git show HEAD:e2e/package.json | grep ...` and a repo-wide `git grep` for a jest/vitest manifest entry) to the verification log, with their actual output, so the Playwright/no-Jest-Vitest claims are directly supported rather than inferred from a directory listing alone.
2. **L390 process-substitution masking risk** — replaced `done < <(find ...)` with an explicit checked temporary file (`find ... > "$list_file"` then `done < "$list_file"`), since `set -e` does not propagate a `find` failure across a process-substitution boundary and could previously mask a partial failure behind a passing non-empty-list check.

Both fixes verified locally (markdownlint-cli2, heuristic lint, shell-snippet lint, and the manifest queries themselves all pass/return expected output). Both threads resolved.

**Comment by lhpaul** [comment]
@coderabbitai review

File: docs/specs/developments/20260811002516_1441-i18n-lint-rule-port/2_1441-i18n-lint-rule-port_implementation-plan.md [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+     done
+     list_file="$(mktemp)"
+     trap 'rm -f "$list_file"' EXIT
+     find docs/specs/developments/20260811002516_1441-i18n-lint-rule-port docs/testing/workflow -name "*.md" -print0 > "$list_file"
+     md_files=()
+     while IFS= read -r -d '' f; do
+       md_files+=("$f")
+     done < "$list_file"
+     [ "${#md_files[@]}" -gt 0 ] || { echo "no markdown files found to lint" >&2; exit 1; }
+     python3 scripts/lint/markdown-heuristic-lint.py CHANGELOG.md "${md_files[@]}"
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/specs/developments/20260811002516_1441-i18n-lint-rule-port/2_1441-i18n-lint-rule-port_implementation-plan.md:395]: Declining this one — CHANGELOG.md is intentionally included as an argument to `markdown-heuristic-lint.py`, not a scope-widening mistake. Per the script's own docstring ("Markdown heuristic lint checks for spec, plan, and CHANGELOG documents") and `scripts/lint/README.md`'s canonical usage example (`find docs/specs/developments docs/testing/workflow -name "*.md" -print0 | xargs -0 python3 scripts/lint/markdown-heuristic-lint.py CHANGELOG.md`), CHANGELOG.md is always passed as an explicit additional argument alongside the directory-scoped files — it is one of the three intended target document types (spec, plan, CHANGELOG), linted in addition to the directory scope, not as part of it. The plan's own 'heuristic lint scope' wording (Testing Strategy scenario 5 and Verification Log) already distinguishes the two directories from CHANGELOG.md; removing CHANGELOG.md would deviate from this repo's own documented convention (confirmed against scripts/lint/README.md and the docstring in scripts/lint/markdown-heuristic-lint.py), not fix a bug. Leaving the command as-is.
